server/etcdserver/api/etcdhttp/health.go

```diff
@@ -365,3 +367,10 @@ func serializableReadCheck(srv ServerHealth) func(ctx context.Context) error {
 		return nil
 	}
 }
+
+func readIndexCheck(srv ServerHealth) func(ctx context.Context) error {
+	return func(ctx context.Context) error {
+		_, err := srv.GetReadIndex(ctx)
```

Comment:
Changing a critical loop like `linearizableReadLoop` seems like a bad idea for backport. What's wrong with directly just calling `requestCurrentIndex`?

Reply:
Just need to have a public function.
```
func (s *EtcdServer) RequestCurrentIndex() (uint64, error) {
  requestId := s.reqIDGen.Next()
  leaderChangedNotifier := s.leaderChanged.Receive()
  retun s.requestCurrentIndex(leaderChangedNotifier, requestId)
}
```

What do you think?

Reply:
ping @siyuanfoundation

Reply:
Oh, I see. `requestCurrentIndex` was designed to be only called from single routine like `linearizableReadLoop`. Nice catch!

Let me think about this more. Need to think on how to minimize the change to `linearizableReadLoop` for a safer backport.
cc @ahrtr for their ideas